pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -370,7 +401,16 @@ func (fs *Decomposedfs) DeleteStorageSpace(ctx context.Context, req *provider.De
 	if err != nil {
 		return err
 	}
-	return nil
+
+	trashPathMatches, err := filepath.Glob(node.InternalPath() + ".T.*")
+	if err != nil {
+		return err
+	}
+	if len(trashPathMatches) != 1 {
+		return fmt.Errorf("delete space failed: found %d matching trashed spaces", len(matches))
+	}
+	trashPath := trashPathMatches[0]
+	return os.Symlink(trashPath, filepath.Join(filepath.Dir(matches[0]), filepath.Base(trashPath)))
```

Comment:
Can you quickly explain why this is needed? Shouldn't the trash be deleted with the space?

Reply:
So before this PR the situation was like this:

Before `DeleteStorageSpace(4c510ada-c86b-4815-8820-42cdf82c3d51)`
```
storage/users/nodes:
2e10cce8-0ed3-45aa-848b-73cb7ff3e68b
4c510ada-c86b-4815-8820-42cdf82c3d51
b4eba12b-0943-4949-8bb7-49e5e185d620
cf19ce0e-d543-4a23-ae06-f16ae10a0fb5
fdaa630e-57ce-4c4d-8463-37901e943af7
root

storage/users/spaces/personal:
4c510ada-c86b-4815-8820-42cdf82c3d51 -> ../../nodes/4c510ada-c86b-4815-8820-42cdf82c3d51
```

After `DeleteStorageSpace(4c510ada-c86b-4815-8820-42cdf82c3d51)`
```
storage/users/nodes:
2e10cce8-0ed3-45aa-848b-73cb7ff3e68b
4c510ada-c86b-4815-8820-42cdf82c3d51.T.{timestamp}
b4eba12b-0943-4949-8bb7-49e5e185d620
cf19ce0e-d543-4a23-ae06-f16ae10a0fb5
fdaa630e-57ce-4c4d-8463-37901e943af7
root

storage/users/spaces/personal:
```

This results in the space not being listed but only soft deleted.
Now since we want to list soft deleted spaces and purge them the line above links the soft deleted space in `storage/users/spaces/personal`. So the result looks like this now.

After `DeleteStorageSpace(4c510ada-c86b-4815-8820-42cdf82c3d51)`
```
storage/users/nodes:
2e10cce8-0ed3-45aa-848b-73cb7ff3e68b
4c510ada-c86b-4815-8820-42cdf82c3d51.T.{timestamp}
b4eba12b-0943-4949-8bb7-49e5e185d620
cf19ce0e-d543-4a23-ae06-f16ae10a0fb5
fdaa630e-57ce-4c4d-8463-37901e943af7
root

storage/users/spaces/personal:
4c510ada-c86b-4815-8820-42cdf82c3d51.T.{timestamp} -> ../../4c510ada-c86b-4815-8820-42cdf82c3d51.T.{timestamp}
```

Reply:
:+1:  Thanks!